cnm.go

```diff
@@ -16,29 +16,147 @@
 
 package virtcontainers
 
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/01org/ciao/ssntp/uuid"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/pkg/types"
+)
+
 // cnm is a network implementation for the CNM plugin.
 type cnm struct {
 	config NetworkConfig
 }
 
+func (n *cnm) createResult(addr net.Addr) (types.Result, error) {
+	var res types.Result
+	res.IP4.IP.IP = net.ParseIP(addr.String())
+
+	return res, nil
+}
+
+func (n *cnm) createEndpointsFromScan() ([]Endpoint, error) {
+	var endpoints []Endpoint
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return []Endpoint{}, err
+	}
+
+	uniqueID := uuid.Generate().String()
+
+	for i, addr := range addrs {
+		var endpoint Endpoint
+
+		if addr.Network() == "lo" || reflect.DeepEqual(net.ParseIP(addr.String()), net.IPv4(127, 0, 0, 1)) {
+			continue
+		} else {
```

Comment:
But I don't want to go through the code below (endpoint.Properties, err = n.createResult(addr)) in case I find the "lo" interface.